### ISSUE TITLE
Concurrency support

### ISSUE THREAD

**Issue by mattmassicotte** [michalbencur/SwiftyJSCore#5]
Hello! I stumbled across this package and it looks, more or less, like exactly what I was hoping to find. Thanks for making it!

I was glancing through the code, and I saw some stuff that surprised me around concurrency usage. I discovered that complete checking is not enabled. Some of the design decisions may make it hard to get this code working warning-free. But, I'd be interested in helping! Is that something that sounds interesting to you?

Here's how I turned it on:

```diff
diff --git a/Package.swift b/Package.swift
index 49aea91..8dd699f 100644
--- a/Package.swift
+++ b/Package.swift
@@ -20,10 +20,12 @@ let package = Package(
     targets: [
         .target(
             name: "SwiftyJSCore",
-            dependencies: []),
+            dependencies: [],
+            swiftSettings: [.enableExperimentalFeature("StrictConcurrency")]),
         .testTarget(
             name: "SwiftyJSCoreTests",
             dependencies: ["SwiftyJSCore"],
-            resources: [.process("script.js")]),
+            resources: [.process("script.js")],
+            swiftSettings: [.enableExperimentalFeature("StrictConcurrency")]),
     ]
 )
```

**Comment by michalbencur** [comment]
Sure, contributions are welcome.
I have't look at latest Swift features for some time, so I'm not sure what issues there are. Out of curiosity I tried to apply these settings to see what warnings I get, but nothing really changed.

**Comment by mattmassicotte** [comment]
With it turned on, there are problems around assuming that `JSValue` is Sendable. It may *become* sendable in a future OS version though! And Swift 6's region-based isolation may help.

There are also issues around `JSInterpreter.setupFetch`, which I had a look at, and will be challenging to adapt...

**Comment by michalbencur** [comment]
Based on few evenings studying swift concurrency I updated SwiftyJSCore and just released 0.3.0. My application using it now compiles and runs fine with Swift 6.

I'm getting errors when using `Any` types as arguments (parameters to my JS functions), I guess the API should be changed from using `Any` to `Sendable`. I'm not sure about it though and let `Any` in for compatibility reasons.
